### PR TITLE
fix: release notes categories

### LINKED ISSUE #127: Release notes categories
**Describe the bug**
Automatically generated release notes rely on PR-labels and not titles to assign them to a specific category. PR issues should be retroactively tagged and a github action on labeling correctly based on PR title should be added so that the user does not need to do this manually

**To Reproduce**
1. Make new release
2. Generate notes
3. The keywords for the PR-title set in .github/release.yml are not recognised

### PR DESCRIPTION
**Pull request by mkatsanto**
## Description

- New labels have been manually added in the repo, called enhancement and breaking-change. 
- A new github action labeller attaches appropriate label according to PR-title
- The .github/release.yml has been modified to accept the approved label names to assign PRs to appropriate categories

Fixes #127

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## Conventional Commits guidelines

- [X] I made sure the PR title follows the 
https://www.conventionalcommits.org/en/v1.0.0/

Changes to workflow inputs (sample table and/or configs)
* major (add **BREAKING CHANGE:** in the beginning of the PR title)
    * more fields/properties are required 
    * existing ones are dropped entirely 
* minor (add **feat:** in the beginning of the PR title)
    * optional fields/properties are added
    * required ones are made optional

Changes to workflow outputs
* major (add **BREAKING CHANGE:** in the beginning of the PR title)
    * changes lead to removal/change of existing outputs (format or location)
* minor (add **feat:** in the beginning of the PR title)
    * additional outputs are generated
    * content (but not format or location) of existing outputs changes

Everything else: patch
(add any other conventional commit in the beginning of the PR title)

## Checklist:

- [X] My code changes follow the style of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] I have updated the project's documentation

### CODE REVIEW

**Comment by mkatsanto** [comment]
tested the behaviour in a fork and it worked, but had to give read write access to the workflow. Not sure if this needs to be set here as well.

**Review by uniqueg — CHANGES_REQUESTED** [review]
The described behavior/expectations for breaking changes do _not_ conform to the Conventional Commits specification, which stipulate that breaking changes should either be indicated by an exclamation mark (`!`) following the type/scope of a commit (just before the colon), the inclusion of the term `BREAKING CHANGE:` in the commit message footer, or both. As it is difficult to correctly set commit message footers by way of squashed commits in feature branches (PR titles cannot have multiple lines and it is easy to forget to adapt the commit message and format it properly according to Conventional Commits when squash merging branches), I would recommend to only support the first syntax, i.e., the inclusion of an exclamation mark. To be clear, including `BREAKING CHANGE` in the title of a commit is violating the Conventional Commits standard!

**Comment by mkatsanto** [comment]
thanks for the comment @uniqueg, currently the label making relies only on the PR title and not the commit name, can you explain a bit more what you mean with the including "BREAKING CHANGE in the title of a commit?"

**Review by uniqueg — APPROVED** [review]

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: 1
+labels:
+- label: "breaking-change"
+  title: "BREAKING CHANGE:.*"
```
Comment (dominikburri) [inline .github/labeler.yml:4]: Should this not be in the footer, in contrast to the title specified here? 
See https://www.conventionalcommits.org/en/v1.0.0/ 

Also, the label is duplicated, would this cause issues?

**Pull request merged**